docs/metrics-yaml-format.md

```diff
@@ -279,3 +285,17 @@ observation:
     objectName: solr/${core}:type=searcher,*
 
 ```
+
+## How to group metrics in YAML file
+Typically multiple metrics are grouped under single YAML file. 
+
+* DB data source - all metrics from single DB query can be grouped under one YAML file. For example, in Clickhouse 
+  integration all profile event metrics (can be fetched using single query) are grouped under 
+  [db-system-events.yml](https://github.com/sematext/sematext-agent-integrations/blob/master/clickhouse/db-system-events.yml).
+* JSON data source - all metrics from single URL can be grouped under one YAML file. For example, in Elasticsearch
+  integration all indexing related stats that can be fetched from single URL is grouped under 
+  [json-index-1plus.yml](https://github.com/sematext/sematext-agent-integrations/blob/master/elasticsearch/json-index-1plus.yml) 
```

Comment:
In case of ES we extract metrics from same URL across multiple YAML files. So we have the same URL source for json-index-1plus.yml, json-merge-1plus.yml etc. I'd say in single YAML file all observation must use the same URL (it is just how our config structure is), but single URL can be used in multiple different YAML files which are logically separated by what kind of metrics are collected. One important thing to mention is probably also that using same URL in N different YAML files doesn't mean agent will send N exactly the same requests - instead, agent caches requests/responses internally, so during single collection, all N YAML files using same URL actually produce just 1 request to ES.